src/Diagnostics.RuntimeHost/Services/DevOpsClient/DevOpsClient.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                List<GitCommitRef> commits = await gitClient.GetCommitsAsync(_project, _repoID, query);
+                return commits[0].CommitId;
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+        }
+
+        public async Task<object> GetBranchesAsync(string resourceUri)
+        {
+            object result = null;
+
+            try
+            {
+                List<GitBranchStats> branches = await gitClient.GetBranchesAsync(_project, _repoID);
+                result = branches.Select(x => x.Name).ToList();
+            }
+            catch(Exception ex)
+            {
+                DiagnosticsETWProvider.Instance.LogDevOpsApiException(
+                    resourceUri,
+                    ex.Message,
+                    ex.StackTrace
+                    );
+                throw new VssServiceException();
+            }
+
+            return result;
+        }
+        
+        public async Task<object> GetFileContentAsync(string filePathInRepo, string resourceUri, string branch = null)
+        {
+            object result = null;
+            GitVersionDescriptor version = null;
+
+            try
+            {
+                if (!string.IsNullOrWhiteSpace(branch))
+                {
+                    version = new GitVersionDescriptor()
+                    {
+                        Version = branch,
+                        VersionType = GitVersionType.Branch
+                    };
+                }
+
+                GitItem item = await gitClient.GetItemAsync(_project, _repoID, path: filePathInRepo, includeContent: true, versionDescriptor: version);
+                result = item.Content;
+            }
+            catch(Exception ex)
+            {
+                DiagnosticsETWProvider.Instance.LogDevOpsApiException(
+                    resourceUri,
+                    ex.Message,
+                    ex.StackTrace
+                    );
+                throw new VssServiceException();
+            }
+
+            return result;
+        }
+
+        public async Task<object> MakePullRequestAsync(string sourceBranch, string targetBranch, string title, string resourceUri)
+        {
+            GitPullRequest pr = new GitPullRequest();
+            object result = null;
+            string source = $"refs/heads/{sourceBranch}";
+            string target = $"refs/heads/{targetBranch}";
+            pr.SourceRefName = source;
+            pr.TargetRefName = target;
+            pr.Title = title;
+
+            try
+            {
+                result = await gitClient.CreatePullRequestAsync(pr, _project, _repoID);
+            }
+            catch (Exception ex)
+            {
+                DiagnosticsETWProvider.Instance.LogDevOpsApiException(
+                    resourceUri,
+                    ex.Message,
+                    ex.StackTrace
+                    );
+                throw new VssServiceException();
```

Comment:
Do we need to throw a `new VssServiceException()` here? Can we just rethrow the original exception? This may end up changing the stack trace when its captured by RuntimeHost's default unhandled exception event logging.